Added CI/CD considerations for private Nuget feeds (#5729)
umbraco/UmbracoDocs: umbraco-cloud/set-up/project-settings/umbraco-cicd/KnownLimitationsAndConsiderations.md

**Usage of Private NuGet Feeds**
* If you have followed the [documentation](https://docs.umbraco.com/umbraco-cloud/set-up/private-nuget-feed) to use a private NuGet feed on Umbraco Cloud, you need to make sure that your NuGet.config file has the correct credentials on your leftmost environment. 
* If your leftmost environment does not have the credentials, and you try to push the NuGet.config changes through CI/CD, you will encounter problems.
* You can add the correct credentials that reference your Umbraco Cloud secrets by following the steps below:
  1. Login to Kudu on your leftmost environment.
  2. Navigate to your NuGet.config file via the console (the default location is the repository root).
  3. Edit the NuGet.config file to include your private feed and credentials.
  4. Save the file.
  5. Commit the changes while in Kudu.
* After making sure that your NuGet.config has the credentials for your private feed, copy the NuGet.config file to your local project. You should be able to trigger the CI/CD flow successfully afterwards.

> 

Reviewer: I feel like it makes for an easier read, if point (*) 1, 2 and 4 are simply normal sentences without the (*) :)

Author: done :)

> As we continue to prototype and gather insights from beta users, there are some known limitations and considerations to be aware of:

Reviewer: ```suggestion
As we continue to gather insights from users, there are some known limitations and considerations to be aware of:
```

Author: done :)

* Given the necessity to avoid changes in other environments, the lack of strict coordination among multiple teams or individuals working on the same project elevates the risk of conflicts.

> **Usage of Private NuGet Feeds**

Reviewer 2: as this limitation is quite big and has some steps to follow I suggest to add it as a sub-section under this "## Potential Limitations and Considerations" section. 

```suggestion
###Usage of Private NuGet Feeds
```

It is possible to reference private NuGet feeds [by using the Secrets Management feature](../../private-nuget-feed.md) on Umbraco Cloud. 

> If you plan on using this feature, the first time you are setting up the private feed configuration has to be done manually. The guide describes how to correctly set it up. The initial setup is not possible through the CI/CD pipeline.
> 
> This is because Umbraco Cloud collects the private NuGet credentials from your leftmost environment. If your leftmost environment does not have the credentials, trying to push the NuGet.config changes through CI/CD will cause restore problems.

Reviewer 2: it makes better sense to have the text "The initial setup is not possible through the CI/CD pipeline." in 2nd paragraph as there is where an explanation why it is not working is 👍 
```suggestion
If you plan on using this feature, the first time you are setting up the private feed configuration has to be done manually. The guide describes how to correctly set it up. 

The initial setup is not possible through the CI/CD pipeline. This is because Umbraco Cloud collects the private NuGet credentials from your [left-most environment](https://docs.umbraco.com/umbraco-cloud/deployments/deployment). If your left-most environment doesn't have the credentials, then it will cause restore problems when trying to push the NuGet.config changes through CI/CD.
```

> The initial setup is not possible through the CI/CD pipeline. This is because Umbraco Cloud collects the private NuGet credentials from your [left-most environment](https://docs.umbraco.com/umbraco-cloud/deployments/deployment). If your left-most environment doesn't have the credentials, then it will cause restore problems when trying to push the NuGet.config changes through CI/CD.

Reviewer 2: ```suggestion
The initial setup is not possible through the CI/CD pipeline. This is because Umbraco Cloud collects the private NuGet credentials from your [left-most environment](https://docs.umbraco.com/umbraco-cloud/deployments/deployment). If your left-most environment doesn't have the credentials, then it will cause restore problems when trying to push the `NuGet.config` changes through CI/CD.
```

You can also add the correct credentials that reference your Umbraco Cloud secrets by following the steps below:

>   1. Login to Kudu on your leftmost environment.

Reviewer 2: ```suggestion
  1. Login to [Kudu](https://docs.umbraco.com/umbraco-cloud/set-up/power-tools) on your left-most environment.
```

>   2. Navigate to your NuGet.config file via the console (the default location is the repository root).

Reviewer 2: ```suggestion
  2. Navigate to your `NuGet.config` file via the console (the default location is the repository root).
```

>   3. Edit the NuGet.config file to include your private feed and credentials.

Reviewer 2: ```suggestion
  3. Edit the `NuGet.config` file to include your private feed and credentials.
```

  4. Save the file.
  5. Commit the changes while in Kudu.

> After making sure that your NuGet.config has the credentials for your private feed, copy the NuGet.config file to your local project. You should be able to trigger the CI/CD flow successfully afterwards.

Reviewer 2: ```suggestion
After making sure that your `NuGet.config` has the credentials for your private feed, copy the `NuGet.config` file to your local project. You should be able to trigger the CI/CD flow successfully afterward.
```

> After making sure that your NuGet.config has the credentials for your private feed, copy the NuGet.config file to your local project. You should be able to trigger the CI/CD flow successfully afterwards.

Reviewer 2: at least in support, we never recommend out clients to make any changes in Kudu so they do not get used to making those changes there and then after a push from local or previous environment for the changes to be overwritten/removed as they were not committed. Wouldn´t it better in this case to use the same steps as https://docs.umbraco.com/umbraco-cloud/set-up/private-nuget-feed#step-4-add-private-myget-feed-on-umbraco-cloud to just make this change to the  NuGet.config file  locally and push it up to cloud environment? 

@jesp209i can I have your opinion on this as well?
